java/statement/Label.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+package graql.lang.statement;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nullable;
+import java.io.Serializable;
+
+/**
+ * A Label is composed of a Name and an optional Scope
+ */
+public class Label implements Comparable<Label>, Serializable {
+    private static final long serialVersionUID = 2051578406740868932L;
+
+    private final String name;
+    private final String scope;
+    private final String scopedName;
+
+    public Label(String name, @Nullable String scope) {
+        if (name == null) {
+            throw new NullPointerException("Null value");
+        }
+        this.name = name;
+        this.scope = scope;
+        if (scope == null) {
+            scopedName = name;
+        } else {
+            scopedName = scope + ":" + name;
+        }
+    }
+
+    /**
+     * @param value The string which potentially represents a Type
+     * @return The matching Type Label
+     */
+    @CheckReturnValue
+    public static Label of(String value, @Nullable String scope) {
+        return new Label(value, scope);
+    }
+
+    public String name() {
```

Comment:
Also question here whether this name() method should return the full name WITH scope, or the default non-scoped name